src/main/java/org/candlepin/guice/CandlepinContextListener.java

```diff
@@ -250,6 +254,9 @@ private void destroySubsystems() throws Exception {
             }
         }
 
+        EventAdapter eventAdapter = injector.getInstance(EventAdapter.class);
```

Comment:
Are we certain the scope and lifecycle of this needs to be managed by us at this level? I would expect that the provider of such a thing would be handling that transparently.

Reply:
I'm not sure I understand what you mean.  Unless I am missing something, CandlepinContextListener seems to be the only place to perform logic during application startup and shutdown. I don't believe (unless I was missing something) Guice provides a way for hook into the lifecycle of classes in comparison to Spring with the `PostConstruct` and `PreDestroy` annotations.

Reply:
So, the purpose of these lifecycle events is to establish and close the `javax.jms.Connection` which is essentially a TCP connection with SSL when using the `ActiveMQSslConnectionFactory` class to establish the connection.  This is based on the `javax.jms.Connection` documentation:

https://docs.oracle.com/javaee%2F7%2Fapi%2F%2F/javax/jms/Connection.html
> It encapsulates an open connection with a JMS provider. It typically represents an open TCP/IP socket between a client and the service provider software.

So, failure to close the TCP connection on our application shutdown could result in our client leaking inactive connection and consuming from the ActiveMQ maximumConnections count.

Changes made:
- I updated CandlepinContextListener to only initialize and shutdown one instance of `EventAdapter`.
- I updated `EventAdapter.startup` to `EventAdapter.initialize`.